bt-core/src/main/java/bt/data/file/FileSystemStorageUnit.java

```diff
@@ -33,32 +35,35 @@
     private boolean init(boolean create) {
 
         if (closed) {
-            if (!parent.exists()) {
-                if (create && !parent.mkdirs()) {
-                    throw new BtException("Failed to create file storage -- can't create (some of the) directories");
-                }
+        	if (!Files.exists(parent)) {
+        		try {
+        			Files.createDirectory(parent);
```

Comment:
You should probably use `Files.createDirectories(Path path)`, since it creates all required ancestors, like the `File#mkdirs()` method it is replacing.